src/constants.rs

```diff
@@ -0,0 +1,42 @@
+//! Global project constants.
+
+extern crate directories;
+
+use self::directories::ProjectDirs;
+use std::path::{Path, PathBuf};
+
+/// Path constants like the GC root directory.
+pub struct Paths {
+    gc_root_dir: PathBuf,
+    daemon_socket_file: PathBuf,
+}
+
+impl Paths {
+    /// Set up project paths
+    pub fn new() -> std::io::Result<Paths> {
```

Comment:
Rust's coding style makes `new` never return errors. Maybe `get` or `build` or `initialize` is a better choice here?